server/service/src/invoice/customer_return/update/generate.rs

```diff
@@ -112,24 +112,31 @@ fn set_new_status_datetime(customer_return: &mut InvoiceRow, patch: &UpdateCusto
     let current_datetime = Utc::now().naive_utc();
 
     match (&customer_return.status, new_status) {
-        // From New/Picked/Shipped to Delivered
+        // From New/Picked/Shipped to Received (Always skip Delivered for Customer Returns)
         (
             InvoiceStatus::New | InvoiceStatus::Picked | InvoiceStatus::Shipped,
```

Comment:
One more case here, to 'Delivered'

Reply:
Yes, I agree! I changed to that in this PR as I think this approach is prone to errors...